pkgs/games/osu-lazer/default.nix

```diff
@@ -69,14 +70,14 @@ in stdenv.mkDerivation rec {
     dotnet publish osu.Desktop \
       --no-build \
       --configuration Release \
+      --runtime ${runtimeId} \
       --no-self-contained \
       --output $out/lib/osu
-    shopt -s extglob
-    rm -r $out/lib/osu/runtimes/!(${runtimeId})
 
     makeWrapper $out/lib/osu/osu\! $out/bin/osu\! \
       --set DOTNET_ROOT "${dotnet-netcore}" \
-      --suffix LD_LIBRARY_PATH : "${lib.makeLibraryPath runtimeDeps}"
+      --suffix LD_LIBRARY_PATH : "${lib.makeLibraryPath runtimeDeps}" \
+      --add-flags --tk # FIXME: Default SDL2 has cursor issues.
```

Comment:
Can we make this flag optional (with default to true)? This way, it would be easier to debug this SDL issue (and also allow someone to force SDL/Tk if they want).

Reply:
BTW, I tested without this flag and seems to work fine for me. I am using `i3wm`, are you using a Wayland desktop maybe @oxalica ?

Reply:
@thiagokokada 
> Can we make this flag optional (with default to true)?

I'm not sure how to make an enable-by-default optional flag. Maybe check an environment variable like `OSU_NO_DEFAULT_FLAGS`?

> Are you using a Wayland desktop?

No. I'm using KDE with almost default settings (X currently).
```nix
  services.xserver = {
    enable = true;
    layout = "us";
    dpi = 144;
    desktopManager.plasma5.enable = true;
    displayManager.sddm.enable = true;
  };
```

Reply:
> I'm not sure how to make an enable-by-default optional flag. Maybe check an environment variable like OSU_NO_DEFAULT_FLAGS?

I was thinking something like this:
https://github.com/NixOS/nixpkgs/blob/b83e5dbc3cd164481cb556e6a5d1c0451d8bd950/pkgs/servers/http/unit/default.nix#L47-L57

So we could have a flag, `withTk`, that when set would do something like this:

```nix
''
    makeWrapper $out/lib/osu/osu\! $out/bin/osu\! \
      --set DOTNET_ROOT "${dotnet-netcore}" \
      --suffix LD_LIBRARY_PATH : "${lib.makeLibraryPath runtimeDeps}" \
" ++ with optional withTk --add-flags --tk # FIXME: Default SDL2 has cursor issues.
```

But well, if it is working with SDL2 now I don't see a reason to complicate.